sklearn/preprocessing/label.py

```diff
@@ -126,12 +126,15 @@ def transform(self, y):
         """
         check_is_fitted(self, 'classes_')
         y = column_or_1d(y, warn=True)
+        # transform of empty array is empty array
+        if _num_samples(y) == 0:
```

Comment:
I think once we've run column_or_1d, we should have an array, and `len` or `.shape[0]` should be sufficient, rather than the more generic _num_samples..?

Reply:
> using len==0 will let some odd objects to pass e.g. using transform(" ") will have len(" ") = 1 which will result in error.

I think we can use `.shape[0]` and it's a better choice here. `column_or_1d` can get rid of the situation mentioned above. We're using `_num_samples` because previously there's isn't `column_or_1d` in `inverse_transform`. @jnothman do you think we need a patch for this?